src/functions.js

```diff
@@ -128,6 +173,7 @@ function factorial( int ) {
 
   // calculate and return the factorial of int
   // note: factorial of 0 is 1
+  return (int <= 1)?1:factorial(int - 1)* int;
```

Comment:
Wow, some recursion! I would be careful with that, especially as it's not always the most readable solution 🙂